Duke.java

```diff
@@ -0,0 +1,216 @@
+                    count++;
+                    System.out.println("Now you have " + String.valueOf(taskArrayList.size()) + " tasks in the list.");
+                } else if (response.startsWith("deadline ") && response.length() > 9) {
+                    String description = response.substring(9, response.indexOf("/by")).trim();
+                    String by = response.substring(response.indexOf("/by") + 4).trim();
+                    Deadline task = new Deadline(description, count, by);
+                    taskArrayList.add(task);
+                    task.addTask(description);
+                    count++;
+                    System.out.println("Now you have " + String.valueOf(taskArrayList.size()) + " tasks in the list.");
+                } else if (response.startsWith("delete ") && response.length() > 7) {
+                    int id = Integer.valueOf(response.substring(7));
+                    Task task = taskArrayList.get(id - 1);
+                    taskArrayList.remove(id - 1);
+                    System.out.println("Noted. I've removed this task:");
+                    System.out.println(task.toString().substring(2));
+                    System.out.println("Now you have " + String.valueOf(taskArrayList.size()) + " tasks in the list.");
+                    for (int i = id; i < taskArrayList.size() - 1; i++) {
+                        Task temp = taskArrayList.get(i - 1);
+                        temp = new Task(temp.description, i - 2);
+                    }
+                }
+                else {
+                    //taskArrayList.add(new Task(response, count));
+
+                    CustomException exception = new CustomException();
+                    throw exception;
+                }
+            } catch (CustomException exception) {
+                System.out.println("☹ OOPS!!! The description of a todo cannot be empty.");
+            } catch (StringIndexOutOfBoundsException exception) {
+                System.out.println("☹ OOPS!!! The description lacks details.");
+            }
+        }
+        scanner.close();
+    }
+
+
+
+
+    private void saveTasksToFile() {
+        File file = new File(filePath);
+        File directory = file.getParentFile();
+
+        try {
+            // Create the directory if it doesn't exist
+            if (!directory.exists()) {
+                directory.mkdirs();
+            }
+
+            // Create the file if it doesn't exist
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+
+            try (FileOutputStream fos = new FileOutputStream(file);
+                 ObjectOutputStream oos = new ObjectOutputStream(fos)) {
+                // Write the taskArrayList to the file
+                oos.writeObject(taskArrayList);
+            } catch (IOException e) {
+                System.out.println("Error saving tasks to file!");
+                e.printStackTrace();
+            }
```

Comment:
I think you can modify this inner try catch block to follow the coding standards as instructed on website!
![image](https://github.com/nus-cs2103-AY2324S1/ip/assets/99804772/74483167-6524-4320-908b-3be5eccce5b1)